recirq/time_crystals/dtcexperiment.py

```diff
@@ -0,0 +1,144 @@
+        self.g = g
+
+        if qubits is None:
+            qubit_locations = [(3, 9), (3, 8), (3, 7), (4, 7), (4, 8), (5, 8), (5, 7), (5, 6),
+                    (6, 6), (6, 5), (7, 5), (8, 5), (8, 4), (8, 3), (7, 3), (6, 3)]
+            self.qubits = [cirq.GridQubit(*idx) for idx in qubit_locations]
+        else:
+            self.qubits = qubits
+
+        num_qubits = len(self.qubits)
+
+        if initial_states is None:
+            self.initial_states = np.random.choice(2, (self.disorder_instances, num_qubits))
+        else:
+            self.initial_states = initial_states
+
+        if local_fields is None:
+            self.local_fields = np.random.uniform(-1.0, 1.0, (self.disorder_instances, num_qubits))
+        else:
+            self.local_fields = local_fields
+
+        zero_params = [thetas, zetas, chis]
+        for index, zero_param in enumerate(zero_params):
+            if zero_param is None:
+                zero_params[index] = np.zeros((self.disorder_instances, num_qubits - 1))
+            else:
+                zero_params[index] = zero_param
+        self.thetas, self.zetas, self.chis = zero_params
+
+        # if gamma or phi is not supplied, generate it from the other such that phis == -2*gammas
+        if gammas is None and phis is None:
+            self.gammas = -np.random.uniform(0.5*np.pi, 1.5*np.pi,
+                                (self.disorder_instances, num_qubits - 1))
+            self.phis = -2*self.gammas
+        elif phis is None:
+            self.gammas = gammas
+            self.phis = -2*self.gammas
+        elif gammas is None:
+            self.phis = phis
+            self.gammas = -1/2*self.phis
+        else:
+            self.phis = phis
+            self.gammas = gammas
+
+    def param_resolvers(self) -> cirq.Zip:
+        """ return a sweep over disorder instances for the parameters of this experiment
+        Returns:
+            `cirq.Zip` object with self.disorder_instances many `cirq.ParamResolver`s
+        """
+
+        # initialize the dict and add the first, non-qubit-dependent parameter, g
+        factor_dict = {'g': np.full(self.disorder_instances, self.g).tolist()}
+
+        # iterate over the different parameters
+        qubit_varying_factors = ["initial_states", "local_fields", "thetas",
+                                    "zetas", "chis", "gammas", "phis"]
+        for factor in qubit_varying_factors:
+            parameter = getattr(self, factor)
+            # iterate over each index in the qubit chain and the various options for that qubit
+            for index, qubit_factor_options in enumerate(parameter.transpose()):
+                factor_name = factor[:-1]
+                factor_dict[f'{factor_name}_{index}'] = qubit_factor_options.tolist()
+
+        return cirq.study.dict_to_zip_sweep(factor_dict)
+
+def comparison_experiments(qubits: Sequence[cirq.Qid],
+                            disorder_instances: int,
```

Comment:
weird formatting nit: the level of indentation is not aligned here. Consider using an automated code formatter like the built-in pycharm one or `black`